cli/remotebazel/remotebazel.go

```diff
@@ -167,27 +170,66 @@ func determineRemote(repo *git.Repository) (*git.Remote, error) {
 }
 
 func determineDefaultBranch(repo *git.Repository) (string, error) {
-	branches, err := repo.Branches()
-	if err != nil {
-		return "", status.UnknownErrorf("could not list branches: %s", err)
+	branch, localErr := localDefaultRef(repo)
+	if localErr == nil {
+		return branch, nil
 	}
 
-	allBranches := make(map[string]struct{})
-	err = branches.ForEach(func(branch *plumbing.Reference) error {
-		allBranches[string(branch.Name())] = struct{}{}
-		return nil
-	})
+	log.Warnf("Could not determine default git branch locally. Finding the default branch on the 'origin' remote.\n")
+	branch, remoteErr := remoteDefaultRef(repo)
+	if remoteErr == nil {
+		return branch, nil
+	}
+
+	return "", status.NotFoundErrorf("could not determine default branch: %s", errors.Join(localErr, remoteErr))
+}
+
+func localDefaultRef(repo *git.Repository) (string, error) {
+	var localErr error
+	// Check for existence of
+	//   refs/heads/main
+	//   refs/heads/master
+	for _, defaultRefName := range []plumbing.ReferenceName{plumbing.Master, plumbing.Main} {
```

Comment:
did you want to use `defaultRefNames` here instead of `[]plumbing.ReferenceName{...}`?